### PR TITLE
Load chart data

### PR DESCRIPTION
**Pull request by bbrewer-papa-x**

### CODE REVIEW

File: src/store/query-selectors.js [hunk 1]
```diff
@@ -3,14 +3,15 @@ import { isEmpty } from 'lodash'
...
+const queryData = state => state.queryReducer.queryData
 
 export const getQueryFailureMessage = state => state.queryReducer.queryFailureMessage
 export const getQueryIsLoading = state => state.queryReducer.isQueryLoading
 
 export const userHasInteracted = state => state.queryReducer.userInteracted
 
-export const getQueryData = state => state.queryReducer.queryData || []
-export const isQueryDataAvailable = state => state.queryReducer.queryData !== null
+export const getQueryData = createSelector(queryData, queryData => queryData || [])
```
Comment (bbrewer-papa-x) [inline src/store/query-selectors.js]: NOTE: this was to remove several extra renders as the `[]` was technically a "new" value

File: src/store/reducers/query-reducer.test.js [hunk 2]
```diff
@@ -62,5 +61,29 @@ describe("queryReducer", () => {
...
+    expect(newState.queryHasBeenExecuted).toBe(true)
   })
-})
\ No newline at end of file
+
+  it('resetQuery resets to the default state', () => {
+    const currentState = {
+      queryText: "select * from lucky_charms",
+      queryData: ['stuff'],
+      queryFailureMessage: "oh noooooo",
```
Comment (ManApart) [inline src/store/reducers/query-reducer.test.js]: lol

File: src/store/sagas/visualization-saga.test.js [hunk 3]
```diff
@@ -68,128 +82,141 @@ describe('visualization-saga', () => {
...
+          AuthenticatedHTTPClient.post = jest.fn(() => ({ status: 200, data: returnedVisualization }))
+          dispatched = await recordSaga(saveVisualization, visualizationSave({title, query}), initialState)
+        })
+
+        it('calls api with parameters that include a dereferenced chart', () => {
+          expect(AuthenticatedHTTPClient.post).toHaveBeenCalledWith(`/api/v1/visualization`,
+            {
+              title: title,
+              query: query,
+              chart: { data: { x: [], xsrc: "col1" }, frames: [], layout: {} } // TODO: Double check the expected x
```
Comment (ManApart) [inline src/store/sagas/visualization-saga.test.js]: Is this comment temporary? (Aka going to live forever?)

File: src/store/sagas/visualization-saga.js [hunk 4]
```diff
@@ -8,7 +8,20 @@ function* loadVisualizationSaga() {
...
+      yield put(setChartInformation(response.data.chart))
+      yield put(setQueryText(response.data.query))
+      yield put(executeFreestyleQuery(response.data.query))
+      yield put(visualizationLoadSuccess(response.data))
+    } else {
+      yield put(visualizationLoadFailure(response.status))
+    }
+  } catch (e) {
+    yield put(visualizationLoadFailure(e.message))
+  }
```
Comment (cameronmarsh) [inline src/store/sagas/visualization-saga.js]: It's no big deal, but would it make sense to extract this to its own function to be consistent with `saveVisualization`?

File: src/store/sagas/visualization-saga.test.js (hunk 3 above)
Comment (cameronmarsh) [inline src/store/sagas/visualization-saga.test.js]: I see this TODO is already there, but can we do something to get rid of that

Comment (cameronmarsh) [reply src/store/sagas/visualization-saga.test.js]: pretty sure it should be `null`

Comment (mcsearchin) [reply src/store/sagas/visualization-saga.test.js]: Nope... we didn't add it - probably just the result of a refactor.  We'll remove it.  Thanks.

File: src/store/sagas/visualization-saga.js (hunk 4 above)
Comment (mcsearchin) [reply src/store/sagas/visualization-saga.js]: The part of @bbrewer-papa-x that loves symmetry likes this idea.  We think it's a bit premature though.

File: src/store/sagas/visualization-saga.test.js (hunk 3 above)
Comment (mcsearchin) [reply src/store/sagas/visualization-saga.test.js]: Fixed.  Thanks.

**Pull request merged**